packages/core-mobile/app/utils/getJsonRpcErrorMessage/getJsonRpcErrorMessage.ts

```diff
@@ -34,6 +85,13 @@ export const parseJsonRpcError = (
 
   // Handle detailed error message based on cause
   if (cause?.error?.message) {
+    // Try to get Solana specific error message
+    const solanaError = getSolanaErrorMessage(cause.error.message)
+    if (solanaError) {
+      return solanaError
+    }
```

Comment:
we should probably start thinking about how to handle rpc error message better since what we have right now is not really scalable. maybe we can include the chain id + action (send/swap/etc) and narrow down the logic accordingly.